core/CMakeLists.txt

```diff
@@ -15,21 +15,31 @@ add_library(prometheus-cpp-core
 
 set_target_properties(prometheus-cpp-core PROPERTIES EXPORT_NAME core)
 
-target_link_libraries(prometheus-cpp-core PRIVATE ${CMAKE_THREAD_LIBS_INIT})
-if(UNIX AND NOT APPLE)
-  target_link_libraries(prometheus-cpp-core PRIVATE rt)
-endif()
+target_link_libraries(prometheus-cpp-core
+  PRIVATE
+    ${CMAKE_THREAD_LIBS_INIT}
+    $<$<AND:$<BOOL:UNIX>,$<NOT:$<BOOL:APPLE>>>:rt>
+)
 
-target_include_directories(prometheus-cpp-core PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
+target_include_directories(prometheus-cpp-core
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
 
-install(TARGETS prometheus-cpp-core EXPORT prometheus-cpp-targets
+install(
+  TARGETS prometheus-cpp-core
+  EXPORT prometheus-cpp-targets
```

Comment:
For consistency `${PROJECT_NAME}-targets`? But I'd find the `prometheus-cpp-targets` more readable.